src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -26,6 +29,42 @@
  */
 class CacheClearCommand extends ContainerAwareCommand
 {
+    private $filesystem;
+    private $cacheClearer;
+    private $cacheDir;
+
+    /**
+     * @param Filesystem            $filesystem
+     * @param CacheClearerInterface $cacheClearer
+     * @param string                $cacheDir
+     */
+    public function __construct($filesystem = null, CacheClearerInterface $cacheClearer = null, $cacheDir = null)
+    {
+        parent::__construct();
+
+        if (!$filesystem instanceof Filesystem) {
```

Comment:
What about using `is_dir($baseDir)` to differentiate a command name from the base dir, if we make it the first constructor arg?
The baseDir is already supposed to exist. The command will throw an exception otherwise.